src/plugins/Miner.cpp

```diff
@@ -73,6 +80,38 @@ void CallDownMULE() {
     }
 }
 
+void DistrubuteMineralWorker(const sc2::Unit* unit_) {
+    if (!unit_)
+        return;
+
+    sc2::Point2D worker_loc = unit_->pos;
+    sc2::Point2D target_loc = gAPI->observer().StartingLocation();
+    float target_dist = std::numeric_limits<float>::max();
+
+    for (auto& i : gHub->GetExpansions()) {
+        if (i.owner != Owner::SELF)
+            continue;
+
+        const sc2::Unit* th = gAPI->observer().GetUnit(i.town_hall_tag);
+        float dist = DistanceSquared2D(worker_loc, th->pos);
+
+        if (dist < target_dist && th->build_progress == 1.0f
+            && th->assigned_harvesters < th->ideal_harvesters) {
+            target_dist = dist;
+            target_loc = th->pos;
+        }
+    }
+
+    auto patches = gAPI->observer().GetUnits(
+        sc2::IsVisibleMineralPatch(), sc2::Unit::Alliance::Neutral);
+    const sc2::Unit* mineral_target = patches.GetClosestUnit(target_loc);
```

Comment:
This part is not resolved. Previously I suggested to keep mineral patches in Expansion class.

Reply:
I started implementing this, but I've run into an issue where it seems the Snapshot Tag#'s for minerals stored in Expansions during GameStart somehow change Tag# when used later, and do not return a valid Unit when calling GetUnit. I've been troubleshooting it for a couple hours, and give up for now. I've moved it to #56 "Save minerals/geysers in Expansions". I'll push it as a separate branch/PR so you can see what I have so far.

Reply:
This is a known issue: when we try to search for expansions under the fog of war, we don’t get snapshots tags, not real one. This is how the client works.
However, the api has a special event (not handled here) which notifies when we first time see a unit. We should check whether this is a mineral and update the tags list.

Reply:
Is it okay to wait to refactor this in the different PR? Currently, this is called infrequently.

Reply:
Yep, this is ok, too.